addon-tools/compare-to-helm/README.md

```diff
@@ -0,0 +1,148 @@
+but the best use case is to include the minimal number of values in order to build the template. This tool adds some features
+that can help in finding the minimum values needed in order to build the CRs from the templates.
+It's important to mention that the user is in charge of completing the values.yaml, because it includes his own preferences/configuration.
+
+example:
+
+```yaml
+DirName_TemplateName:
+- apiVersion: v1
+  metadata:
+    name: defaultName  
+    namespace: default
+sa: # template in same dir as metadata.yaml 
+- apiVersion: v2
+  metadata:
+    name: example
+DirName_DirName2_secret:
+- metadata:   # values for first CR that will be created from  DirName/DirName2/secret.yaml template 
+    name: name
+- metadata:  # values for second CR that will be created from  DirName/DirName2/secret.yaml template 
+    name: name2
+```
+
+helm convertor creates a helm chart where each template in the kube-compare formatted reference includes a corresponding template in the helm chart.
+The helm convertor wraps each template in an additional `range` defined in go templating syntax, the range is defined to loop
+over the CRs values as defined in the chart values. This results in a creation of a CR for each element in the corresponding
+slice in the values.yaml.
+
+## Easy Creation of values.yaml
+
+creating a values.yaml from scratch may be very frustrating to do without any automation.
+This can be complicated for references including multiple templates, because it requires going over all the templates,
+and gathering a list of which values are required to be able to build the CR.
+
+To address this problem the Tool creates a values.yaml file structure that contains values that are required in order
+to build the CRs. Then the user only needs to fill in the values.yaml file with the default values without needing to
+think about which values are needed and without needing to build the structure.
+The list of values that are needed are collected by a traversal on the static form of template.
+Due to the big amount of possibles in go templating syntax some cases where values are used can be not detected.
+This version covers all the basic use-cases, More complex cases can be added by demand. Additionally, different
+approaches to this issue can be added.  
+The initial implementation of the values gathering is a great place to start with, and for complex cases users may will
+need to so some minor modifications. This i also a one time thing per reference, for references that have already been converted
+the previous values.yaml can be used as a base for the values.yaml file. Most of the work is in the initial conversion.
+
+example generated values.yaml structure (does not include the default values):
+
+```yaml
+#gnerated from: pkg/compare/testdata/AllRequiredTemplatesExistAndThereAreNoDiffs/reference/metadata.yaml
+cm:
+- {} # Template with no user defined content  
```

Comment:
Is a template has no used defined data why not just skip it.

Reply:
yeah, your right changed it